src/common/cockpitjsonprint.c

```diff
@@ -183,7 +183,13 @@ FILE *
 cockpit_json_print_open_memfd (const char *name,
                                int         version)
 {
-  int fd = memfd_create ("cockpit login messages", MFD_ALLOW_SEALING | MFD_CLOEXEC);
+  /* current kernels moan about not specifying exec mode */
+#ifdef MFD_NOEXEC_SEAL
+  int fd = memfd_create ("cockpit login messages", MFD_ALLOW_SEALING | MFD_CLOEXEC | MFD_NOEXEC_SEAL);
```

Comment:
Having the variable definition inside of the `#ifdef` is probably not going to work out...

Reply:
Eww, yes -- sorry, I only tested on F39 locally, but the tests were screaming at me indeed. Fixed.